_pytest/python.py

```diff
@@ -889,13 +892,22 @@ def parametrize(self, argnames, argvalues, indirect=False, ids=None,
         if scope is None:
             scope = "function"
         scopenum = scopes.index(scope)
+        valtypes = {arg: "funcargs" for arg in argnames}
         if not indirect:
             #XXX should we also check for the opposite case?
             for arg in argnames:
                 if arg not in self.fixturenames:
                     raise ValueError("%r uses no fixture %r" %(
                                      self.function, arg))
-        valtype = indirect and "params" or "funcargs"
+        else:
+            if not isinstance(indirect, (tuple, list)):
+                valtypes = {arg: "params" for arg in argnames}
```

Comment:
It seems dict comprehensions are invalid syntax in Python 2.6, which is why the checks are failing. You can probably do something like `dict([(arg, "params") for arg in argnames])` instead (but I didn't test it :wink:).